source/vulkan/vulkan_hooks.cpp

```diff
@@ -454,14 +693,399 @@ VkResult VKAPI_CALL vkQueuePresentKHR(VkQueue queue, const VkPresentInfoKHR *pPr
+	createInfo.subresourceRange.aspectMask = VK_IMAGE_ASPECT_DEPTH_BIT;
+
+	VkImageView pViewReplacement = VK_NULL_HANDLE;
+	result = trampoline(device, &createInfo, nullptr, &pViewReplacement);
+	if (result != VK_SUCCESS)
+	{
+		LOG(WARN) << "> vkCreateImageView failed with error code " << result << '!';
+		return result;
+	}
+
+	// Guard access to the map
+	const std::lock_guard<std::mutex> lock(s_mutex);
+
+	const auto it = s_depth_stencil_buffer_images.find(createInfo.image);
+	if (it != s_depth_stencil_buffer_images.end())
+		s_depth_stencil_buffer_imageViews[*pView] = { it->first, createInfo, it->second, pViewReplacement };
+#endif
+
+	return result;
+}
+
+VkResult VKAPI_CALL vkCreateFramebuffer(VkDevice device, const VkFramebufferCreateInfo* pCreateInfo, const VkAllocationCallbacks* pAllocator, VkFramebuffer* pFramebuffer)
+{
+	PFN_vkCreateFramebuffer trampoline = nullptr;
+
+	{ const std::lock_guard<std::mutex> lock(s_mutex);
+		trampoline = s_device_dispatch.at(get_dispatch_key(device)).CreateFramebuffer;
+		assert(trampoline != nullptr);
+	}
+
+	const VkResult result = trampoline(device, pCreateInfo, pAllocator, pFramebuffer);
+	if (result != VK_SUCCESS)
+	{
+		LOG(WARN) << "> vkCreateFramebuffer failed with error code " << result << '!';
+		return result;
+	}
+
+	// Guard access to the map
+	const std::lock_guard<std::mutex> lock(s_mutex);
+
+	const VkImageView *pAttachments = pCreateInfo->pAttachments;
+	for (uint32_t i = 0; i < pCreateInfo->attachmentCount; i++)
+	{
+		VkImageView imageView = pAttachments[i];
+
+		const auto it = s_depth_stencil_buffer_imageViews.find(imageView);
+		if (it != s_depth_stencil_buffer_imageViews.end())
+			s_depth_stencil_buffer_frameBuffers[*pFramebuffer] = { it->first, it->second };
+	}
+
+	return result;
+}
+
+VkResult VKAPI_CALL vkAllocateCommandBuffers(VkDevice device, const VkCommandBufferAllocateInfo* pAllocateInfo, VkCommandBuffer* pCommandBuffers)
+{
+	PFN_vkAllocateCommandBuffers trampoline = nullptr;
+
+	{ const std::lock_guard<std::mutex> lock(s_mutex);
+		trampoline = s_device_dispatch.at(get_dispatch_key(device)).AllocateCommandBuffers;
+		assert(trampoline != nullptr);
+	}
+
+	const VkResult result = trampoline(device, pAllocateInfo, pCommandBuffers);
+	if (result != VK_SUCCESS)
+	{
+		LOG(WARN) << "> vkAllocateCommandBuffers failed with error code " << result << '!';
+		return result;
+	}
+
+	// Guard access to the map
+	const std::lock_guard<std::mutex> lock(s_mutex);
+
+	for (uint32_t i = 0; i < pAllocateInfo->commandBufferCount; i++)
+	{
+		VkCommandBuffer commandBuffer = pCommandBuffers[i];
+		s_command_buffer_mapping[commandBuffer] = device;
+
+		reshade::vulkan::draw_call_tracker command_buffer_tracker;
+		add_command_buffer_trackers(commandBuffer, command_buffer_tracker);
+	}
+
+	return result;
+}
+
+VkResult VKAPI_CALL vkCreateRenderPass(VkDevice device, const VkRenderPassCreateInfo* pCreateInfo, const VkAllocationCallbacks* pAllocator, VkRenderPass* pRenderPass)
```

Comment:
This too does not seem to do anything, so probably don't need to hook it either.

Reply:
OK. Done